### PR TITLE
Better polling logic

### PR DESCRIPTION
**Pull request by AaronVasquez**

### CODE REVIEW

File: src/upload.js [hunk 1]
```diff
@@ -52,7 +52,9 @@ export const EVENTS = Object.freeze({
    * @property {Number} progress.progress - the percentage of progress for the chunk upload
    */
   CHUNKS_PROGRESS: "chunks-progress",
-  RETRIEVED: "retrieved", // Maybe change this to "uploaded", with upload-progress renamed attach-progress or something
+  UPLOADED: "uploaded",
+  META_ATTACHED: "meta-attached", // Same as RETRIEVED
```
Comment (CKH4) [inline src/upload.js]: Can you document this event?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "oyster-streamable",
-  "version": "2.1.5",
+  "version": "2.2.1",
```
Comment (EdmundMai) [inline package.json]: why not 2.1.6 or 2.2.0

File: src/upload.js [hunk 3]
```diff
@@ -52,7 +52,9 @@ export const EVENTS = Object.freeze({
...
+  RETRIEVED: "retrieved", // DEPRECATED
```
Comment (EdmundMai) [inline src/upload.js]: can we remove this if it's depracated?

Comment (EdmundMai) [reply src/upload.js]: o nvm if it's still being used (just saw below)

File: package.json (hunk 2 above)
Comment (AaronVasquez) [reply package.json]: Its a minor update, and I already published to 2.2.0 to QA

**Pull request merged**